lib/email/receiver.rb

```diff
@@ -171,8 +171,20 @@ def process_internal
 
         raise first_exception if first_exception
 
-        if post = find_related_post(force: true)
-          if Guardian.new(user).can_see_post?(post) && post.created_at < 90.days.ago
```

Comment:
@ZogStriP I wanted to get your thoughts on this. If I'm understanding our requirements correctly, we only want to raise a `OldDestinationError` when the PostReplyKey is invalid. The assumption here is would mean that an invalid `PostReplyKey` means old though.

Reply:
Hmm I realized that the assumption where if we can't find a `PostReplyKey` means that the destination is old. Reverted in https://github.com/discourse/discourse/pull/6289/files#diff-0327bfc5fb2dd2a7fd635a096e7fb0daR187

Reply:
I don't think we need to specifically check for an invalid post reply key. If the user takes too much time to reply via email, just gently tell them to visit the forum.

Reply:
@ZogStriP Yea I think we can try that too. I think I'm a little overzealous in getting it 100% accurate for post replies via email.